tickets/router.py

```
@@ -1,27 +1,225 @@
+            now = datetime.utcnow()
+            next_pending = draws_db.get_next_pending_draw(req.category_id)
+            if next_pending and next_pending.scheduled_open_time <= now:
+                update_res = draws_db.update_draw(next_pending.id, DrawUpdateSchema(status="open", actual_open_time=now))
+                if update_res:
+                    target_draw = draws_db.get_draw_by_id(next_pending.id)
+                else:
+                    raise HTTPException(status_code=500, detail="Failed to open a pending draw for ticket purchase.")
+
+            if not target_draw:
+                if category.draw_interval_type == "manual":
+                     raise HTTPException(status_code=400, detail=f"Category {category.name} is manual. No open draw available and new draws must be created manually.")
```

Flake8 (7.2.0)

[error] indentation is not a multiple of 4

(E111)